### DISCUSSION TITLE
Compile-to-js compatibility

### DISCUSSION THREAD

**Discussion by wokalski** [concrete-utopia/utopia#1392] [Q&A]
We're using Rescript in our project, others might use something like Purescript or maybe Flow. What's the compatibility story with javascript flavors. Is it simple to add support for a new one?

**Comment by balazsbajorics** [comment]
hi there! 

We have two types of source code: what we consider "first party", these are the files that can be directly manipulated using the canvas/inspector and you can see the source code updated. This first release of Utopia only supports javascript with JSX. Our focus for this year is to make our Javascript/JSX support better and work out the kinks before we think about extending to other languages.

However, we also support importing code from node_modules and other "third party" code. Our only requirement here is that the compiled code should be browser-compatible javascript. A good example is a component library you import from npm, you can place and use the components on the screen, but you can't make changes to the component library itself.
For code in node_modules, we have a set of loaders that are currently hardwired. But it would be possible to extend this system to support other flavors of compile-to-js code in a non-interactive fashion.
